src/commands/cork_proposal/cellar_v2_5/set_share_price_oracle.rs

```diff
@@ -0,0 +1,86 @@
+use abscissa_core::{clap::Parser, Command, Runnable};
+
+use crate::{
+    application::APP,
+    cellars::{self, is_evm_address},
+    commands::cork_proposal::print_proposal,
+    prelude::*,
+    proto::{
+        cellar_v2_5governance::{Function, SetSharePriceOracle},
+        governance_call::Call,
+        CellarV25governance, GovernanceCall,
+    },
+    utils::string_to_u256,
+};
+
+#[derive(Command, Debug, Parser)]
+#[clap(
+    long_about = "DESCRIPTION\n\nCalls setSharePriceOracle() on the target V2.5 cellar contract at the specified block height.\nFor more information see https://github.com/PeggyJV/cellar-contracts/blob/main/src/base/Cellar.sol"
+)]
+pub struct SetSharePriceOracleCmd {
+    #[clap(short, long)]
+    /// ID of the oracle registry
+    registry_id: String,
+
+    #[clap(short, long)]
+    /// Address of the oracle
+    oracle_address: String,
```

Comment:
Nit: apologies for the repetition, but just tagging the places where it's different.